packages/contracts/tests/python/test.py

```diff
@@ -2708,6 +2713,106 @@ def test_is_history_of_unfinalized_question_valid(self):
+
+        self._advance_clock(40)
+        self.assertTrue(self.rc0.functions.isFinalized(early_question_id).call())
+        self.assertEqual(self.rc0.functions.resultFor(early_question_id).call(), to_answer_for_contract(1001))
+
+        txid = self.rc0.functions.askQuestionERC20(
+            0,
+            "my question",
+            self.arb0.address,
+            30,
+            0,
+            0
+            ,1100
+        ).transact(self._txargs())
+        question_id = calculate_question_id(self.rc0.address, 0, "my question", self.arb0.address, 30, 0, 0, self.web3.eth.accounts[0], 0)
+
+        # Start with the series we used before and make sure it goes through
+        st = None
+        st = self.submitAnswerReturnUpdatedState( st, question_id, 1001, 0, 200, k3)
+        st = self.submitAnswerReturnUpdatedState( st, question_id, 1001, 200, 400, k3)
+
+        freezeTs = self.rc0.functions.freezeTs().call()
+        self.assertTrue(freezeTs == 0, "Freeze timestamp should be zero before freeze")
+        self.rc0.functions.setFreezeTimestamp(self._block_timestamp()).transact(self._txargs())
+        freezeTs = self.rc0.functions.freezeTs().call()
+        self.assertTrue(freezeTs > 0, "Freeze timestamp should be not zero after freeze")
+
+        # You can no longer answer a question
+        with self.assertRaises(TransactionFailed):
+            self.submitAnswerReturnUpdatedState( st, question_id, 1001, 400, 800, k3)
+
+        # You can no longer request arbitration
+        fee = self.arb0.functions.getDisputeFee(encode_hex("0x00")).call()
+        txid = self.arb0.functions.requestArbitration(question_id, 0).transact(self._txargs(val=fee))
+        with self.assertRaises(TransactionFailed):
+            self.raiseOnZeroStatus(txid)
+
+        # The question will no longer finalize, even after the time has elapsed
+        self.assertFalse(self.rc0.functions.isFinalized(question_id).call())
+        self._advance_clock(40)
+        self.assertFalse(self.rc0.functions.isFinalized(question_id).call())
+        # Can't get result as not finalized
+        with self.assertRaises(TransactionFailed):
+            self.assertEqual(self.rc0.functions.resultFor(question_id).call(), to_answer_for_contract(1001))
+
+        # The question that was finalized before is still finalized
+        self.assertTrue(self.rc0.functions.isFinalized(early_question_id).call())
+        # You can still get the result
+        self.assertEqual(self.rc0.functions.resultFor(early_question_id).call(), to_answer_for_contract(1001))
+
+        # You can no longer ask a new question
+        with self.assertRaises(TransactionFailed):
+            txid = self.rc0.functions.askQuestionERC20(
+                0,
+                "my question xyz",
+                self.arb0.address,
+                30,
+                0,
+                0
+                ,1100
+            ).transact(self._txargs())
+            self.raiseOnZeroStatus(txid)
```

Comment:
you could add tests that claiming and withdrawing is also frozen.

Reply:
Done in [710abe9](https://github.com/RealityETH/reality-eth-monorepo/pull/147/commits/710abe9e5200a899833f308b04cb6adf47a0b1aa)